reverse_file_buffered.py

```diff
@@ -0,0 +1,38 @@
+from file_io import Io, FileIo
+
+
+def reverse_file_in_memory(io: Io, name: str, buf_size: int):
+    """Reverse a content of the file `name` using no more than buf_size of additional memory"""
+    assert buf_size >= 2, "Minimum allowed buf_size is 2"
+
+    length = io.file_length(name)
+    chunk = buf_size//2
+    output_buff = bytearray(chunk*2)
+    mid = length//2
+    left_offset = 0
+    right_offset = length - left_offset - min(chunk, mid - left_offset)
+
+
+    while left_offset < right_offset or 0 < min(chunk, mid - left_offset) < chunk:
+        if length <= chunk:
+            chunk = length
+            output_buff = bytearray(chunk)
+            io.read_file(name, left_offset, chunk, output_buff, 0)
+            reversed_output_buff = output_buff[::-1]
+            io.write_file(reversed_output_buff, 0, chunk, name, left_offset)
+            break
+
+        io.read_file(name, left_offset, chunk, output_buff, 0)
```

Comment:
In the loop, you never update `chunk` size. Thus, if at some point left_offset and right_offset are closer than `chunk*2`, but further than `chunk`, you will load and swap overlapping chunks.

You can implement the code easier, if you introduce a helper function, that just swaps 2 chunks given the left offset, right offset, and chunk size. Then in the loop you can focus on properly calculating these numbers